### DISCUSSION TITLE
Question regarding specific automatic reprocessing of messages based on destination response

### DISCUSSION THREAD

**Discussion by ghost** [nextgenhealthcare/connect#4574] [Q&A] [answered]
Is there a way I can reprocesses a message a specific amount of times for a specific error ? 
For example if a destination returns a timeout error I'd like it to be reprocessed at least double the amount of times a normal error is reprocessed. My ideal case would be to just keep reprocessing the timeout errors ad infinitum but only those.

**Comment by jonbartels** [comment, marked as answer]
Yes there is.

Check out post 9 from narupley on the old forum thread at https://forums.mirthproject.io/forum/mirth-connect/support/16942-

That example shows how to write a response transformer to queue and retry a message. For your use case you would also want to inspect the content of `msg` (that should be the error returned) and queue/error as you need to.

Note that you will often have to set your response datatype to `RAW`. When the datatype is `RAW` the response connector always runs. If it is some other datatype and the response is not that exact type, the connector will not run.

**Reply by ghost** [reply]
Thanks a lot. I don't know how I missed that.
